sdk/mobile-center-updates/src/main/java/com/microsoft/azure/mobile/updates/Updates.java

```diff
@@ -160,9 +165,14 @@
     private DownloadTask mDownloadTask;
 
     /**
-     * Current task to process download completion.
+     * Current task to check download state and act on it.
      */
-    private ProcessDownloadCompletionTask mProcessDownloadCompletionTask;
+    private CheckDownloadTask mCheckDownloadTask;
+
+    /**
+     * Remember if we checked download since our own process restarted.
+     */
+    private boolean mCheckedDownload;
```

Comment:
If the process restarts, you don't know if last time the process crashes while processing the download manager intent. So I need a in memory flag to do that check once per process.